src/Core__Array.res

```diff
@@ -57,10 +57,10 @@ let equal = (a, b, eq) => {
 
 let rec compareFromIndex = (a, b, i, cmp, len) =>
   if i === len {
-    0
+    Core__Ordering.equal
   } else {
     let c = cmp(a->getUnsafe(i), b->getUnsafe(i))
-    if c === 0 {
+    if c->Core__Ordering.isEqual {
```

Comment:
If we do `c === Core__Ordering.equal` here instead then we save one function call and have less changes in the JS output. Same in Core_List.

Reply:
Done in https://github.com/rescript-association/rescript-core/pull/149/commits/13c38da56fb32fda4dda00c0f7814ab2833b8c80